pkg/network/ebpf/c/protocols/protocol-classification-defs.h

```diff
@@ -17,7 +17,16 @@
 
 // Postgres
 
-#define POSTGRES_MIN_MSG_SIZE 5
+// The minimum size we want to be able to check for a startup message. This size includes:
+// - The length field: 4 bytes
+// - The protocol major version: 2 bytes
+// - The protocol minior version: 2 bytes
+// - The "user" string, as the first connection parameter name: 5 bytes
+#define POSTGRES_STARTUP_MIN_LEN 13
+
+#define PG_STARTUP_VERSION 196608
```

Comment:
what's is that version?
what will happen if there is a newer/older version of postgres?

Reply:
> what's is that version?

That's the version of the protocol, as a 32-bit big-endian number and used in the startup message. Here we check for version `3.0` of the protocol. The magic value comes straight from the protocol specification.

>  what will happen if there is a newer/older version of postgres?

Postgres Protocol 3.0 was introduced with Postgres 8.0 in 2005, and has not changed since. Every currently supported version of Postgres uses it.